src/services/validator.service.ts

```diff
@@ -217,6 +221,89 @@ class ValidatorService {
     };
     return pdcmError;
   }
+
+  #calculateModelScore(sheetsValidationResults: SheetValidationResult[], validationDictionary: SchemasDictionary, data: Map<string, SheetData>){
+    const reportStatus = this.#getUnifiedStatus(sheetsValidationResults)
+    const modelScores: { [model_id: string]: number } = {};
+    if (reportStatus === 'invalid') {
+      return modelScores;
+
+    }
+    
+    // Build weights dictionary
+    const weightsDictionary: { [schemaName: string]: Array<{ fieldName: string; fieldWeight: number; weightForModelType: string }> } = {};
+    validationDictionary.schemas.forEach(schema => {
+      schema.fields.forEach(field => {
+        const { meta } = field;
+        if ( meta && 'field_weight' in meta && 'weight_for_model_type' in meta && typeof meta.field_weight === 'number' && meta?.field_weight > 0 && typeof meta?.weight_for_model_type === 'string') {
```

Comment:
Can we move this validation to a helper method so the conditional is not that long? Something like `if (scoreMetadataExists(meta))` or something similar?

Reply:
Yes definitely